src/tree-manager/tests/singleSelect.test.js

```diff
@@ -0,0 +1,35 @@
+import test from 'ava'
+import TreeManager from '..'
+
+test('should select the same node if it is selected and then deselected right again', t => {
```

Comment:
Not sure I understand this test. E.g., what is the need for this and why does it matter _only_ in the context of `singleSelect`? Can you help me understand, please?

Reply:
Since singleSelect behaves differently than simpleSelect but is very similar this is to ensure we cover the lines for that.

Reply:
Gotcha! In that case, I think we need to structure the test a bit different - checking/unchecking/checking seems like a generic enough task and not specific to single select. Yet, we're forced to do it here to meet code coverage requirements. I'd say those are probably not the right reasons to write a test. 

I'm not sure what the solution is yet, but I'd look into it to see if we can accomplish it in a sense that conveys the right purpose along with testing the required code paths. If not, we'll add a TODO here and move on for now.

Reply:
I guess this test will get rewritten anyway if we separate it out.

Reply:
Added some more test and a snaphot now